app/src/main/java/com/github/polypoly/app/models/game/GameViewModel.kt

```diff
@@ -39,9 +40,18 @@ class GameViewModel(
 
     private val gameEndedData: MutableLiveData<Boolean> = MutableLiveData(false)
 
+    private val playerStateData: MutableLiveData<PlayerState> = MutableLiveData(PlayerState.INIT)
+
     private val tradeRequestData: MutableLiveData<TradeRequest> = MutableLiveData()
 
+    //used to determine if the player is close enough to a location to interact with it
+    private val MAX_INTERACT_DISTANCE = 10.0 // meters
+
     init {
+        setLoading(true)
+        coroutineScope.launch {
+            gameLoop()
+        }
         viewModelScope.launch {
             listenToTradeRequest()
```

Comment:
This line should be moved to into the `coroutineScope.launch` above as it permits dependency injection when testing

Reply:
done